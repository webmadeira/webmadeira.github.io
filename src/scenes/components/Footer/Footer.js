import React from 'react'
import styled from 'styled-components'

const Container = styled.section`
  border: 1px solid;
  flex-shrink: 0;
`

const Footer = () => (
  <Container>
    <p>Footer!</p>
  </Container>
)

export default Footer
