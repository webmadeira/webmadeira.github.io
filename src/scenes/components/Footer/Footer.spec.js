import React from 'react'
import { shallow } from 'enzyme'

import Footer from './Footer'

describe('Footer component', () => {
  it('must render properly', () => {
    const pastEvents = [
      {
        id: 0,
        name: 'Test event',
        url: 'http://test.url.com/',
      },
    ]
    const social = [
      { id: 1, name: 'test social', url: '/testSocial' },
    ]
    const location = {
      name: 'test name',
      address: 'test address',
      postCode: 'test post code',
    }
    const component = shallow(<Footer
      pastEvents={pastEvents}
      social={social}
      location={location}
    />)

    expect(component).toMatchSnapshot()
  })
})
