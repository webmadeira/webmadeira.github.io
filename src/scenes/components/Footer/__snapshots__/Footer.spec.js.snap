// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    pastEvents={
      Array [
        Object {
          "id": 0,
          "name": "Test event",
          "path": "http://test.url.com/",
        },
      ]
    }
    social={
      Array [
        Object {
          "id": 1,
          "name": "test social",
          "path": "/testSocial",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <styled.h2>
            WEBMADEIRA CONFERENCES
          </styled.h2>
          <styled.span>
            <styled.ul>
              <PastEventItem
                name="Test event"
                path="http://test.url.com/"
              />
              <styled.li>
                <styled.a
                  href="/photos"
                  target="_blank"
                >
                  Photos
                </styled.a>
              </styled.li>
              <styled.li>
                <styled.a
                  href="/videos"
                  target="_blank"
                >
                  Videos
                </styled.a>
              </styled.li>
            </styled.ul>
          </styled.span>
        </styled.div>,
        <styled.div>
          <Styled(styled.h2)>
            SOCIAL
          </Styled(styled.h2)>
          <styled.ul>
            <SocialItem
              name="test social"
              path="/testSocial"
            />
          </styled.ul>
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.h2>
              WEBMADEIRA CONFERENCES
            </styled.h2>,
            <styled.span>
              <styled.ul>
                <PastEventItem
                  name="Test event"
                  path="http://test.url.com/"
                />
                <styled.li>
                  <styled.a
                    href="/photos"
                    target="_blank"
                  >
                    Photos
                  </styled.a>
                </styled.li>
                <styled.li>
                  <styled.a
                    href="/videos"
                    target="_blank"
                  >
                    Videos
                  </styled.a>
                </styled.li>
              </styled.ul>
            </styled.span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "WEBMADEIRA CONFERENCES",
            },
            "ref": null,
            "rendered": "WEBMADEIRA CONFERENCES",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <styled.ul>
                <PastEventItem
                  name="Test event"
                  path="http://test.url.com/"
                />
                <styled.li>
                  <styled.a
                    href="/photos"
                    target="_blank"
                  >
                    Photos
                  </styled.a>
                </styled.li>
                <styled.li>
                  <styled.a
                    href="/videos"
                    target="_blank"
                  >
                    Videos
                  </styled.a>
                </styled.li>
              </styled.ul>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  Array [
                    <PastEventItem
                      name="Test event"
                      path="http://test.url.com/"
                    />,
                  ],
                  <styled.li>
                    <styled.a
                      href="/photos"
                      target="_blank"
                    >
                      Photos
                    </styled.a>
                  </styled.li>,
                  <styled.li>
                    <styled.a
                      href="/videos"
                      target="_blank"
                    >
                      Videos
                    </styled.a>
                  </styled.li>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "function",
                  "props": Object {
                    "name": "Test event",
                    "path": "http://test.url.com/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <styled.a
                      href="/photos"
                      target="_blank"
                    >
                      Photos
                    </styled.a>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Photos",
                      "href": "/photos",
                      "target": "_blank",
                    },
                    "ref": null,
                    "rendered": "Photos",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <styled.a
                      href="/videos"
                      target="_blank"
                    >
                      Videos
                    </styled.a>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Videos",
                      "href": "/videos",
                      "target": "_blank",
                    },
                    "ref": null,
                    "rendered": "Videos",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(styled.h2)>
              SOCIAL
            </Styled(styled.h2)>,
            <styled.ul>
              <SocialItem
                name="test social"
                path="/testSocial"
              />
            </styled.ul>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "SOCIAL",
            },
            "ref": null,
            "rendered": "SOCIAL",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <SocialItem
                  name="test social"
                  path="/testSocial"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "function",
                "props": Object {
                  "name": "test social",
                  "path": "/testSocial",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <styled.h2>
              WEBMADEIRA CONFERENCES
            </styled.h2>
            <styled.span>
              <styled.ul>
                <PastEventItem
                  name="Test event"
                  path="http://test.url.com/"
                />
                <styled.li>
                  <styled.a
                    href="/photos"
                    target="_blank"
                  >
                    Photos
                  </styled.a>
                </styled.li>
                <styled.li>
                  <styled.a
                    href="/videos"
                    target="_blank"
                  >
                    Videos
                  </styled.a>
                </styled.li>
              </styled.ul>
            </styled.span>
          </styled.div>,
          <styled.div>
            <Styled(styled.h2)>
              SOCIAL
            </Styled(styled.h2)>
            <styled.ul>
              <SocialItem
                name="test social"
                path="/testSocial"
              />
            </styled.ul>
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.h2>
                WEBMADEIRA CONFERENCES
              </styled.h2>,
              <styled.span>
                <styled.ul>
                  <PastEventItem
                    name="Test event"
                    path="http://test.url.com/"
                  />
                  <styled.li>
                    <styled.a
                      href="/photos"
                      target="_blank"
                    >
                      Photos
                    </styled.a>
                  </styled.li>
                  <styled.li>
                    <styled.a
                      href="/videos"
                      target="_blank"
                    >
                      Videos
                    </styled.a>
                  </styled.li>
                </styled.ul>
              </styled.span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "WEBMADEIRA CONFERENCES",
              },
              "ref": null,
              "rendered": "WEBMADEIRA CONFERENCES",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <styled.ul>
                  <PastEventItem
                    name="Test event"
                    path="http://test.url.com/"
                  />
                  <styled.li>
                    <styled.a
                      href="/photos"
                      target="_blank"
                    >
                      Photos
                    </styled.a>
                  </styled.li>
                  <styled.li>
                    <styled.a
                      href="/videos"
                      target="_blank"
                    >
                      Videos
                    </styled.a>
                  </styled.li>
                </styled.ul>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    Array [
                      <PastEventItem
                        name="Test event"
                        path="http://test.url.com/"
                      />,
                    ],
                    <styled.li>
                      <styled.a
                        href="/photos"
                        target="_blank"
                      >
                        Photos
                      </styled.a>
                    </styled.li>,
                    <styled.li>
                      <styled.a
                        href="/videos"
                        target="_blank"
                      >
                        Videos
                      </styled.a>
                    </styled.li>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "0",
                    "nodeType": "function",
                    "props": Object {
                      "name": "Test event",
                      "path": "http://test.url.com/",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <styled.a
                        href="/photos"
                        target="_blank"
                      >
                        Photos
                      </styled.a>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Photos",
                        "href": "/photos",
                        "target": "_blank",
                      },
                      "ref": null,
                      "rendered": "Photos",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <styled.a
                        href="/videos"
                        target="_blank"
                      >
                        Videos
                      </styled.a>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Videos",
                        "href": "/videos",
                        "target": "_blank",
                      },
                      "ref": null,
                      "rendered": "Videos",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(styled.h2)>
                SOCIAL
              </Styled(styled.h2)>,
              <styled.ul>
                <SocialItem
                  name="test social"
                  path="/testSocial"
                />
              </styled.ul>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "SOCIAL",
              },
              "ref": null,
              "rendered": "SOCIAL",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <SocialItem
                    name="test social"
                    path="/testSocial"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "function",
                  "props": Object {
                    "name": "test social",
                    "path": "/testSocial",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
